cms/src/components/content/query.ts

```diff
@@ -0,0 +1,131 @@
+import { db, DocType, TagType, type Uuid, type ContentDto, PublishStatus } from "luminary-shared";
+
+export type ContentOverviewQueryOptions = {
+    languageId: Uuid;
+    parentType: DocType.Post | DocType.Tag;
+    tagType?: TagType;
+    orderBy?: "title" | "updatedTimeUtc" | "publishDate" | "expiryDate";
+    orderDirection?: "asc" | "desc";
+    translationStatus?: "translated" | "untranslated" | "all";
+    publishStatus?: "published" | "scheduled" | "expired" | "draft" | "all";
+    pageSize?: number;
+    pageIndex?: number;
+    tags?: Uuid[];
+    search?: string;
+};
+
+async function contentOverviewQuery(options: ContentOverviewQueryOptions) {
+    if (!options.orderBy) options.orderBy = "updatedTimeUtc";
+    if (!options.orderDirection) options.orderDirection = "desc";
+    if (!options.pageSize) options.pageSize = 20;
+    if (!options.pageIndex) options.pageIndex = 0;
+    if (!options.translationStatus) options.translationStatus = "all";
+    if (!options.publishStatus) options.publishStatus = "all";
+
+    const translated = (await db.docs // This may slow down the query if there are many documents, but it is necessary to be able to include and filter on untranslated documents
+        .where({ type: DocType.Content, language: options.languageId })
+        .toArray()) as ContentDto[];
+    const untranslatedByParentId: Uuid[] = [];
+
+    let res = db.docs.orderBy(options.orderBy);
+    if (options.orderDirection == "desc") res = res.reverse();
+
+    return res
+        .filter((doc) => {
+            const contentDoc = doc as ContentDto;
+
+            // Filter documents by type
+            if (!contentDoc.parentId) return false;
+            if (contentDoc.type != DocType.Content) return false;
```

Comment:
Related to my previous comment, why not include this in the query?

Reply:
It is not possible to use a "where" query together with "orderBy". If we would want to sort the data at a later stage (with sortBy), it would mean that we would need to load all the filtered data into memory, defeating the purpose of pagination.
I believe with pagination the performance will not be too bad, and the "filter" logic is filtering out non-ContentDto documents as early as possible to avoid unnecessary processing.